programs/drift/src/math/amm.rs

```diff
@@ -149,17 +149,28 @@ pub fn estimate_best_bid_ask_price(
     precomputed_trade_price: Option<u64>,
     direction: Option<PositionDirection>,
 ) -> DriftResult<(u64, u64)> {
-    let base_spread_u64 = amm.base_spread.cast::<u64>()?;
     let last_oracle_price_u64 = amm.historical_oracle_data.last_oracle_price.cast::<u64>()?;
 
+    // in price units
+    let base_spread_price_u64 = last_oracle_price_u64
+        .safe_mul(amm.base_spread.cast::<u64>()?)?
```

Comment:
can this every be 0?

Reply:
yeah, if its zero then the lower bound of `est_market_spread` is 0